src/icrc1_auction_api.mo

```diff
@@ -176,36 +181,17 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
 
   private func registerAsset_(ledgerPrincipal : Principal, minAskVolume : Nat) : async* R.Result<Nat, RegisterAssetError> {
     let canister = actor (Principal.toText(ledgerPrincipal)) : (actor { icrc1_decimals : () -> async Nat8; icrc1_symbol : () -> async Text });
-    let decimals = try {
-      Nat8.toNat(await canister.icrc1_decimals());
-    } catch (err) {
-      throw err;
-    };
-    let symbol = try {
-      await canister.icrc1_symbol();
-    } catch (err) {
-      throw err;
-    };
-    for ((assetInfo, i) in Vec.items(assets)) {
-      if (Principal.equal(ledgerPrincipal, assetInfo.ledgerPrincipal)) return #err(#AlreadyRegistered(i));
+    let decimalsCall = canister.icrc1_decimals();
+    let symbolCall = canister.icrc1_symbol();
+    let decimals = try { Nat8.toNat(await decimalsCall) } catch (e) throw e;
+    let symbol = try { await symbolCall } catch (e) throw e;
```

Comment:
If we simply re-throw then we can also remove the try-catch:
```
let decimals = Nat8.toNat(await decimalsCall);
let symbol = await symbolCall;
 ```
 An uncaught throw from any of them is propagated upwards.

Reply:
done